cirq-google/cirq_google/workflow/processor_record.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import abc
+import dataclasses
+
+import cirq
+import cirq_google as cg
+from cirq._compat import dataclass_repr
+
+
+class ProcessorRecord(metaclass=abc.ABCMeta):
+    """A serializable record that maps to a particular `cg.engine.AbstractProcessor`."""
+
+    @abc.abstractmethod
+    def get_processor(self) -> 'cg.engine.AbstractProcessor':
+        """Using this classes' attributes, return a unique `cg.engine.AbstractProcessor`
+
+        This is the primary method that descendants must implement.
+        """
+
+    def get_sampler(self) -> 'cirq.Sampler':
+        """Return a `cirq.Sampler` for the processor specified by this class.
+
+        The default implementation delegates to `self.get_processor()`.
+        """
+        return self.get_processor().get_sampler()
+
+    def get_device(self) -> 'cirq.Device':
+        """Return a `cirq.Device` for the processor specified by this class.
+
+        The default implementation delegates to `self.get_processor()`.
+        """
+        return self.get_processor().get_device()
+
+
+@dataclasses.dataclass(frozen=True)
+class EngineProcessorRecord(ProcessorRecord):
+    """A serializable record of processor_id to map to a `cg.EngineProcessor`.
+
+    This class presumes the GOOGLE_CLOUD_PROJECT environment
+    variable is set to establish a connection to the cloud service.
+
+    Args:
+        processor_id: The processor id.
+    """
+
+    processor_id: str
+
+    def get_processor(self) -> 'cg.EngineProcessor':
+        """Return a `cg.EngineProcessor` for the specified processor_id."""
+        engine = cg.get_engine()
+        return cg.EngineProcessor(
+            project_id=engine.project_id,
+            processor_id=self.processor_id,
+            context=engine.context,
+        )
+
+    def get_sampler(self) -> 'cg.QuantumEngineSampler':
+        """Return a `cg.QuantumEngineSampler` for the specified processor_id.
+
+        This implementation hardcodes the `cg.SQRT_ISWAP_GATESET` to construct
+        the sampler until this argument is made optional.
+        """
+        return self.get_processor().get_sampler(cg.SQRT_ISWAP_GATESET)
```

Comment:
Should the default be the CircuitSerializer?  It's more general and can handle the union of the other gatesets (at the cost of being less tested, but please use this to help test it!)

Reply:
oh wow. when was that merged in? will test it out